wallet/storage/src/internal/mod.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+
+use common::chain::OutPoint;
+use utxo::Utxo;
+
+use crate::{
+    schema::{self as db, Schema},
+    TransactionRw, Transactional, WalletStorage, WalletStorageRead, WalletStorageWrite,
+};
+
+mod store_tx;
+pub use store_tx::{StoreTxRo, StoreTxRw};
+
+/// Store for wallet data, parametrized over the backend B
+pub struct Store<B: storage::Backend>(storage::Storage<B, Schema>);
+
+impl<B: storage::Backend> Store<B> {
+    /// Create a new wallet storage
+    pub fn new(backend: B) -> crate::Result<Self> {
+        let mut storage = Self(storage::Storage::new(backend).map_err(crate::Error::from)?);
+        storage.set_storage_version(1)?;
+        Ok(storage)
+    }
+
+    /// Dump raw database contents
+    pub fn dump_raw(&self) -> crate::Result<storage::raw::StorageContents<Schema>> {
+        self.0.dump_raw().map_err(crate::Error::from)
+    }
+
+    /// Collect and return all utxos from the storage
+    #[allow(clippy::let_and_return)]
+    pub fn read_utxo_set(&self) -> crate::Result<BTreeMap<OutPoint, Utxo>> {
+        let db_tx = self.transaction_ro()?;
+        let map = db_tx.0.get::<db::DBUtxo, _>();
+        let res = map.prefix_iter_decoded(&())?.collect::<BTreeMap<_, _>>();
+
+        Ok(res)
```

Comment:
Why did you decide to silence clippy rather than apply the suggestion?

Reply:
This code was taken from `chainstate/storage` and if the suggestion is applied the code won't compile, saying that `map` won't live long enough.

Reply:
This seems to work and may be worth considering:
```suggestion
    pub fn read_utxo_set(&self) -> crate::Result<BTreeMap<OutPoint, Utxo>> {
        let db_tx = self.transaction_ro()?;
        let dbmap = db_tx.0.get::<db::DBUtxo, _>();
        dbmap.prefix_iter_decoded(&()).map(Iterator::collect::<BTreeMap<_, _>>)
```

Reply:
Or even:
```suggestion
    pub fn read_utxo_set(&self) -> crate::Result<BTreeMap<OutPoint, Utxo>> {
        let db_tx = self.transaction_ro()?;
        db_tx.0.get::<db::DBUtxo, _>().prefix_iter_decoded(&()).map(Iterator::collect)
```

Reply:
nice